server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/AwarenessAllocationDecider.java

```diff
@@ -231,4 +238,19 @@ private static Decision debugNoMissingAttribute(String awarenessAttribute, List<
                 CLUSTER_ROUTING_ALLOCATION_AWARENESS_ATTRIBUTE_SETTING.getKey(),
                 Strings.collectionToCommaDelimitedString(awarenessAttributes));
     }
+
+    private static void validateForceAwarenessSettings(Settings forceSettings) {
+        for (Map.Entry<String, Settings> entry : forceSettings.getAsGroups().entrySet()) {
+            final Optional<String> notValues = entry.getValue().keySet().stream().filter(s -> s.equals("values") == false).findFirst();
+            if (notValues.isPresent()) {
```

Comment:
I am not sure I completely see how applying the state fails when an entry without `values` is found. I do understand it makes little sense though, but depending on how it fails, it may be worthwhile ensure it not only has a `values` entry but also that it is non-empty?

Reply:
Yes, if you set something of the form `cluster.routing.allocation.awareness.force.attr.not_values` today then that doesn't cause a failure at application time, we simply ignore it, so it's really just another bit of inadvertent lenience in the same area that we should be catching at validation time. The failure only comes from setting something of the form `cluster.routing.allocation.awareness.force.no_more_dots`.

I think an empty `cluster.routing.allocation.awareness.force.attr.values` is valid, it effectively says that the only values of `attr` are the ones that currently exist in the cluster which is the default behaviour if the setting is absent.

Reply:
Got it now, thanks for the explanation!